solr/core/src/java/org/apache/solr/handler/component/QueryElevationComponent.java

```diff
@@ -1131,4 +1227,333 @@ public int compareTop(int doc) {
       };
     }
   }
+
+  /**
+   * Matches a collection of subsets of type &lt;E&gt;.
+   * <p>
+   * Given a collection of subsets <code>N</code>, finds all the subsets that are contained (ignoring duplicate elements)
+   * by a provided set <code>s</code>.
+   * That is, finds all subsets <code>n</code> in <code>N</code> for which <code>s.containsAll(n)</code>
+   * (<code>s</code> contains all the elements of <code>n</code>, in any order).
+   * <p>
+   * Associates a match value of type &lt;M&gt; to each subset and provides it each time the subset matches (i.e. is
+   * contained by the provided set).
+   *
+   * @param <E> Subset element type.
+   * @param <M> Subset match value type.
+   */
+  protected interface SubsetMatcher<E, M> {
```

Comment:
I don't think there's a point to this interface (with builder) as separate from implementation since there is one and it's highly theoretical to imagine another user provided one.  So lets combine this with the implementation below, keeping the javadocs.  I suppose the name TrieSubsetMatcher is nicest to me but I don't care.

Reply:
This is the trace of a previous version that had two SubsetMatcher. I remove it.